src/xcode/ENA/ENA/Source/Services/HealthCertificateValidation/BoosterNotificationsService.swift

```diff
@@ -33,8 +33,9 @@ class BoosterNotificationsService: BoosterNotificationsServiceProviding {
 		self.validationRulesAccess = validationRulesAccess
 	}
 	
-	func applyRulesForCertificates(certificates: [DigitalCovidCertificateWithHeader],
-		completion: @escaping (Result<ValidationResult, BoosterNotificationRuleValidationError>) -> Void
+	func applyRulesForCertificates(
+		certificates: [DigitalCovidCertificateWithHeader],
+		completion: @escaping (Result<ValidationResult, BoosterNotificationRuleValidationError>?, HealthCertificateValidationError?) -> Void
```

Comment:
Instead of adding an error to completion, please create a new error Enum for BoosterNotificationsService (BoosterNotificationsService) and wrap the `BoosterNotificationRuleValidationError` in one of its cases.
Then the result type if this completion should be `Result<ValidationResult, BoosterNotificationsServiceError>`